🐛 BUG: wrangler publish fail after syncing files, throwing ERROR

### Which Cloudflare product(s) does this pertain to?

Wrangler core

### What version(s) of the tool(s) are you using?

3.22.2

### What version of Node are you using?

16.13.0

### What operating system are you using?

Linux

### Describe the Bug

After executing this commmand (that was working fine until today 03/01/2024)

# wrangler publish --compatibility-date 2024-01-03 --env dev

when wrangler finish file syncing  it raise an ERROR as you may se:

↗️  Done syncing assets
Total Upload: 131.07 KiB / gzip: 30.51 KiB
Uploaded plataforma-clientes-dev (49.25 sec)

✘ [ERROR] Cannot read properties of undefined (reading 'map')



### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

 ⛅️ wrangler 3.22.2
-------------------
▲ [WARNING] `wrangler publish` is deprecated and will be removed in the next major version.

  Please use `wrangler deploy` instead, which accepts exactly the same arguments.


✔ Would you like to help improve Wrangler by sending usage metrics to Cloudflare? … no
Your choice has been saved in the following file: ../root/.wrangler/metrics.json.

  You can override the user level setting for a project in `wrangler.toml`:

   - to disable sending metrics for a project: `send_metrics = false`
   - to enable sending metrics for a project: `send_metrics = true`
Fetching list of already uploaded assets...
Building list of assets to upload...
 = 10.b580cf7424fbce50.cd7cf62fea.js (already uploaded 10.b580cf7424fbce50.js)
 + 10.b580cf7424fbce50.js.86e041501d.map (uploading new version of 10.b580cf7424fbce50.js.map)
 = 161.382866b6904f6c32.17d5d574aa.js (already uploaded 161.382866b6904f6c32.js)
 + 161.382866b6904f6c32.js.a58c3561b5.map (uploading new version of 161.382866b6904f6c32.js.map)
 = 246.8a39bc39a35e46ce.b588619bac.js (already uploaded 246.8a39bc39a35e46ce.js)
 + 246.8a39bc39a35e46ce.js.29f4199ce7.map (uploading new version of 246.8a39bc39a35e46ce.js.map)
 = 381.e3fd9dcea3c34379.e72ba722e5.js (already uploaded 381.e3fd9dcea3c34379.js)
 + 381.e3fd9dcea3c34379.js.d3e82ef9f8.map (uploading new version of 381.e3fd9dcea3c34379.js.map)
 = 3rdpartylicenses.70d9f7cb88.txt (already uploaded 3rdpartylicenses.txt)
 = 507.488f6af2c41f9d11.91e8b00d24.js (already uploaded 507.488f6af2c41f9d11.js)
 + 507.488f6af2c41f9d11.js.e47ed32e4c.map (uploading new version of 507.488f6af2c41f9d11.js.map)
 = 540.813485c1fcef62e7.9fba33cba8.js (already uploaded 540.813485c1fcef62e7.js)
 + 540.813485c1fcef62e7.js.b483d2a2e4.map (uploading new version of 540.813485c1fcef62e7.js.map)
 = 632.888231fec135baad.4fd7133292.js (already uploaded 632.888231fec135baad.js)
 + 632.888231fec135baad.js.2396255e83.map (uploading new version of 632.888231fec135baad.js.map)
 + 673.0f332cbe92ebbce0.3ad83c7d35.js (uploading new version of 673.0f332cbe92ebbce0.js)
 + 673.0f332cbe92ebbce0.js.6e2ad35c1c.map (uploading new version of 673.0f332cbe92ebbce0.js.map)
 = 680.8c022ac238965237.b1ed000868.js (already uploaded 680.8c022ac238965237.js)
 + 680.8c022ac238965237.js.fc47c39ea0.map (uploading new version of 680.8c022ac238965237.js.map)
 = 767.3bb0d934e73046e5.e6c621a939.js (already uploaded 767.3bb0d934e73046e5.js)
 + 767.3bb0d934e73046e5.js.4c98b50915.map (uploading new version of 767.3bb0d934e73046e5.js.map)
 = 769.520257e76891c9e0.7ba11123ae.js (already uploaded 769.520257e76891c9e0.js)
 + 769.520257e76891c9e0.js.41aa8b056c.map (uploading new version of 769.520257e76891c9e0.js.map)
 = 847.8a7218b906aa1ac6.a709c3abe6.js (already uploaded 847.8a7218b906aa1ac6.js)
 + 847.8a7218b906aa1ac6.js.f426bdeecf.map (uploading new version of 847.8a7218b906aa1ac6.js.map)
 = 862.bcbfa099ed37e0cc.0ca9d46e73.js (already uploaded 862.bcbfa099ed37e0cc.js)
 + 862.bcbfa099ed37e0cc.js.b395fc9b8a.map (uploading new version of 862.bcbfa099ed37e0cc.js.map)
 = 870.61a95afd28d051a0.5eca98539b.js (already uploaded 870.61a95afd28d051a0.js)
 + 870.61a95afd28d051a0.js.a0f4a965d5.map (uploading new version of 870.61a95afd28d051a0.js.map)
 = 875.b6543b0aa9ec8bec.dbaa864ffb.js (already uploaded 875.b6543b0aa9ec8bec.js)
 + 875.b6543b0aa9ec8bec.js.88bbd9ec0d.map (uploading new version of 875.b6543b0aa9ec8bec.js.map)
 = 951.80cb357a2808eb96.6155b3612f.js (already uploaded 951.80cb357a2808eb96.js)
 + 951.80cb357a2808eb96.js.492309f0b1.map (uploading new version of 951.80cb357a2808eb96.js.map)
 = 986.ac8cc2d03f318e25.8c50ff7ec6.js (already uploaded 986.ac8cc2d03f318e25.js)
 + 986.ac8cc2d03f318e25.js.778d3c2566.map (uploading new version of 986.ac8cc2d03f318e25.js.map)
 = assets/042081_33.26e0510644.png (already uploaded assets/042081_33.png)
 = assets/042093_15.4f0407079a.png (already uploaded assets/042093_15.png)
 = assets/050486_notificacionoperativaingresodeplata.fb00d4ec51.png (already uploaded assets/050486_notificacionoperativaingresodeplata.png)
 = assets/404-emoji.f9363ab50d.png (already uploaded assets/404-emoji.png)
 = assets/Cuenta-bancaria.9306783922.png (already uploaded assets/Cuenta-bancaria.png)
 + assets/Logo_Balanz-international.0e44ae01f4.svg (uploading new version of assets/Logo_Balanz-international.svg)
 + assets/Logo_Balanz.d09496a9e2.svg (uploading new version of assets/Logo_Balanz.svg)
 = assets/Test-perfil-inversor.0e8787504d.png (already uploaded assets/Test-perfil-inversor.png)
 = assets/WhatsApp.c8290605a3.png (already uploaded assets/WhatsApp.png)
 + assets/alert-circle 2.e488869897.svg (uploading new version of assets/alert-circle 2.svg)
 = assets/arrow-left.3af06ec597.png (already uploaded assets/arrow-left.png)
 = assets/arrow-right.740c0cdd43.png (already uploaded assets/arrow-right.png)
 = assets/bonos/noImage-bonos.07d4bacd74.svg (already uploaded assets/bonos/noImage-bonos.svg)
 = assets/busqueda_no_resultados.22564a9f72.png (already uploaded assets/busqueda_no_resultados.png)
 + assets/cauciones/caucionesdm.ff14b7c0b7.svg (uploading new version of assets/cauciones/caucionesdm.svg)
 + assets/cauciones/caucionespesos.49f98df4b1.svg (uploading new version of assets/cauciones/caucionespesos.svg)
 + assets/charting_library/charting_library.min.d.d34fb3288e.ts (uploading new version of assets/charting_library/charting_library.min.d.ts)
 + assets/charting_library/charting_library.min.0fba48c5c6.js (uploading new version of assets/charting_library/charting_library.min.js)
 + assets/charting_library/datafeed-api.d.f0448efd51.ts (uploading new version of assets/charting_library/datafeed-api.d.ts)
 + assets/charting_library/static/ar-tv-chart.aaac22e21df68f2f7bad.c95fb4d8fd.html (uploading new version of assets/charting_library/static/ar-tv-chart.aaac22e21df68f2f7bad.html)
 + assets/charting_library/static/bundles/22.d344a511955b43dbefcc.b832e1a636.js (uploading new version of assets/charting_library/static/bundles/22.d344a511955b43dbefcc.js)
 + assets/charting_library/static/bundles/33.bf2a2fc98cda3004cc87.24c7cb73a2.js (uploading new version of assets/charting_library/static/bundles/33.bf2a2fc98cda3004cc87.js)
 + assets/charting_library/static/bundles/add-compare-dialog.99e6e22e5d6b137269e9.51c391ecdc.js (uploading new version of assets/charting_library/static/bundles/add-compare-dialog.99e6e22e5d6b137269e9.js)
 + assets/charting_library/static/bundles/change-interval-dialog.90d98fd5c50c9ae968da.1d4db96276.js (uploading new version of assets/charting_library/static/bundles/change-interval-dialog.90d98fd5c50c9ae968da.js)
 + assets/charting_library/static/bundles/chart-bottom-toolbar.1831f202498024e77558.3defe3641a.js (uploading new version of assets/charting_library/static/bundles/chart-bottom-toolbar.1831f202498024e77558.js)
 + assets/charting_library/static/bundles/chart-widget-gui.f7226f20f800bbc6bec0.1d365fafbd.js (uploading new version of assets/charting_library/static/bundles/chart-widget-gui.f7226f20f800bbc6bec0.js)
 + assets/charting_library/static/bundles/confirm-inputs-dialog.e8333025d1520384f361.8b946cd2a5.js (uploading new version of assets/charting_library/static/bundles/confirm-inputs-dialog.e8333025d1520384f361.js)
 + assets/charting_library/static/bundles/confirm-symbol-input-dialog.741858b9ef54b3611d31.7cfebcb47b.js (uploading new version of assets/charting_library/static/bundles/confirm-symbol-input-dialog.741858b9ef54b3611d31.js)
 + assets/charting_library/static/bundles/create-dialog.b388c187425e521c5a70.dbfc3de0fe.js (uploading new version of assets/charting_library/static/bundles/create-dialog.b388c187425e521c5a70.js)
 = assets/charting_library/static/bundles/crosshair.6c091f7d5427d0c5e6d9dc3a90eb2b20.d1fc57cbe8.cur (already uploaded assets/charting_library/static/bundles/crosshair.6c091f7d5427d0c5e6d9dc3a90eb2b20.cur)
 = assets/charting_library/static/bundles/dot.ed68e83c16f77203e73dbc4c3a7c7fa1.9d57ee47d1.cur (already uploaded assets/charting_library/static/bundles/dot.ed68e83c16f77203e73dbc4c3a7c7fa1.cur)
 + assets/charting_library/static/bundles/drawing-toolbar.37fd80f32e27f9adff7c.03a4ca5901.js (uploading new version of assets/charting_library/static/bundles/drawing-toolbar.37fd80f32e27f9adff7c.js)
 + assets/charting_library/static/bundles/ds-property-pages.d4223322c10f0f7857b4.8149e6b139.js (uploading new version of assets/charting_library/static/bundles/ds-property-pages.d4223322c10f0f7857b4.js)
 + assets/charting_library/static/bundles/editobjectdialog.4a71a4d935ef0bc68003.51da6c887b.js (uploading new version of assets/charting_library/static/bundles/editobjectdialog.4a71a4d935ef0bc68003.js)
 = assets/charting_library/static/bundles/eraser.0579d40b812fa2c3ffe72e5803a6e14c.9a29061ee0.cur (already uploaded assets/charting_library/static/bundles/eraser.0579d40b812fa2c3ffe72e5803a6e14c.cur)
 + assets/charting_library/static/bundles/export-data.1f128f60b181cb7e556c.09ebaa6303.js (uploading new version of assets/charting_library/static/bundles/export-data.1f128f60b181cb7e556c.js)
 + assets/charting_library/static/bundles/floating-toolbars.a37519a9a72c97ec0ff1.7d1948e141.js (uploading new version of assets/charting_library/static/bundles/floating-toolbars.a37519a9a72c97ec0ff1.js)
 + assets/charting_library/static/bundles/go-to-date-dialog-impl.46a234f324ca2b224b41.a91f343b49.js (uploading new version of assets/charting_library/static/bundles/go-to-date-dialog-impl.46a234f324ca2b224b41.js)
 = assets/charting_library/static/bundles/grab.bc156522a6b55a60be9fae15c14b66c5.2faa57dfb7.cur (already uploaded assets/charting_library/static/bundles/grab.bc156522a6b55a60be9fae15c14b66c5.cur)
 = assets/charting_library/static/bundles/grabbing.1c0862a8a8c0fb02885557bc97fdafe7.50680a29bf.cur (already uploaded assets/charting_library/static/bundles/grabbing.1c0862a8a8c0fb02885557bc97fdafe7.cur)
 + assets/charting_library/static/bundles/hammerjs.9f017652d0f77438961c.fb33f9b9f4.js (uploading new version of assets/charting_library/static/bundles/hammerjs.9f017652d0f77438961c.js)
 + assets/charting_library/static/bundles/header-toolbar.cf3b92914ff2b79584e6.5ff0257850.js (uploading new version of assets/charting_library/static/bundles/header-toolbar.cf3b92914ff2b79584e6.js)
 + assets/charting_library/static/bundles/ie-fallback-logos.ecdc2f1255ecbff1dc5f.e8f13cc44c.js (uploading new version of assets/charting_library/static/bundles/ie-fallback-logos.ecdc2f1255ecbff1dc5f.js)
 + assets/charting_library/static/bundles/lazy-jquery-ui.15f27a20ada0be559796.1b3adff59e.js (uploading new version of assets/charting_library/static/bundles/lazy-jquery-ui.15f27a20ada0be559796.js)
 + assets/charting_library/static/bundles/lazy-velocity.b369bcf3fef7b3d24234.410e1421c8.js (uploading new version of assets/charting_library/static/bundles/lazy-velocity.b369bcf3fef7b3d24234.js)
 + assets/charting_library/static/bundles/library.3e375d0971e0e9bab5bb37e47bb2248a.87e97bccdb.css (uploading new version of assets/charting_library/static/bundles/library.3e375d0971e0e9bab5bb37e47bb2248a.css)
 + assets/charting_library/static/bundles/library.3e375d0971e0e9bab5bb37e47bb2248a.rtl.64f03f9b31.css (uploading new version of assets/charting_library/static/bundles/library.3e375d0971e0e9bab5bb37e47bb2248a.rtl.css)
 + assets/charting_library/static/bundles/library.b2ee7224c8cc9f809bca.fd1b12762d.js (uploading new version of assets/charting_library/static/bundles/library.b2ee7224c8cc9f809bca.js)
 + assets/charting_library/static/bundles/line-tools-icons.d4bcd6d038b2fea6feed.ad46738114.js (uploading new version of assets/charting_library/static/bundles/line-tools-icons.d4bcd6d038b2fea6feed.js)
 + assets/charting_library/static/bundles/load-chart-layout-dialog.642c73fda2f41fa2fd16.5fd835de52.js (uploading new version of assets/charting_library/static/bundles/load-chart-layout-dialog.642c73fda2f41fa2fd16.js)
 + assets/charting_library/static/bundles/lt-pane-views.f615773c99d12f423e91.f00fe05b3a.js (uploading new version of assets/charting_library/static/bundles/lt-pane-views.f615773c99d12f423e91.js)
 + assets/charting_library/static/bundles/objecttreedialog.b9fa79f3c22b36c514bf.d31a4a2ade.js (uploading new version of assets/charting_library/static/bundles/objecttreedialog.b9fa79f3c22b36c514bf.js)
 + assets/charting_library/static/bundles/propertypagesfactory.42f63d141baef6e836f7.32f91e4af7.js (uploading new version of assets/charting_library/static/bundles/propertypagesfactory.42f63d141baef6e836f7.js)
 + assets/charting_library/static/bundles/restricted-toolset.bbc61138ca4ee5d2fa1c.eddca63629.js (uploading new version of assets/charting_library/static/bundles/restricted-toolset.bbc61138ca4ee5d2fa1c.js)
 + assets/charting_library/static/bundles/series-icons-map.359d3b9b312cf5f56733.a770a9dd4f.js (uploading new version of assets/charting_library/static/bundles/series-icons-map.359d3b9b312cf5f56733.js)
 + assets/charting_library/static/bundles/series-pane-views.1ef988f4c3e8149e0428.07f1b53694.js (uploading new version of assets/charting_library/static/bundles/series-pane-views.1ef988f4c3e8149e0428.js)
 + assets/charting_library/static/bundles/study-market.74fb802b68f3dcd94b91.20ecd94648.js (uploading new version of assets/charting_library/static/bundles/study-market.74fb802b68f3dcd94b91.js)
 + assets/charting_library/static/bundles/study-pane-views.d406bf2e4fc2b39d5015.53dab7ac34.js (uploading new version of assets/charting_library/static/bundles/study-pane-views.d406bf2e4fc2b39d5015.js)
 + assets/charting_library/static/bundles/symbol-info-dialog-impl.00fc03c11b8e921b3437.520682af74.js (uploading new version of assets/charting_library/static/bundles/symbol-info-dialog-impl.00fc03c11b8e921b3437.js)
 + assets/charting_library/static/bundles/symbolsearch.2971ae2a9c7830395e0c.3a301e2080.js (uploading new version of assets/charting_library/static/bundles/symbolsearch.2971ae2a9c7830395e0c.js)
 + assets/charting_library/static/bundles/take-chart-image-dialog-impl.bea5cbbc7f5966618a0a.edd6d7c29a.js (uploading new version of assets/charting_library/static/bundles/take-chart-image-dialog-impl.bea5cbbc7f5966618a0a.js)
 + assets/charting_library/static/bundles/vendors.4103ca2e65289b14aad5.03a530378c.js (uploading new version of assets/charting_library/static/bundles/vendors.4103ca2e65289b14aad5.js)
 = assets/charting_library/static/bundles/zoom.e21f24dd632c7069139bc47ae89c54b5.53033a5960.cur (already uploaded assets/charting_library/static/bundles/zoom.e21f24dd632c7069139bc47ae89c54b5.cur)
 + assets/charting_library/static/cs-tv-chart.aaac22e21df68f2f7bad.dd6d5e4cde.html (uploading new version of assets/charting_library/static/cs-tv-chart.aaac22e21df68f2f7bad.html)
 + assets/charting_library/static/da_DK-tv-chart.aaac22e21df68f2f7bad.805e487819.html (uploading new version of assets/charting_library/static/da_DK-tv-chart.aaac22e21df68f2f7bad.html)
   (truncating changed assets log, set `WRANGLER_LOG=debug` environment variable to see full diff)
Uploading 270 new assets...
Skipped uploading 586 existing assets.
Uploaded 100% [270 out of 270]
Removing 270 stale assets...  
↗️  Done syncing assets
Total Upload: 131.07 KiB / gzip: 30.51 KiB
Uploaded plataforma-clientes-dev (49.25 sec)

✘ [ERROR] Cannot read properties of undefined (reading 'map')


If you think this is a bug then please create an issue at https://github.com/cloudflare/workers-sdk/issues/new/choose
✔ Would you like to report this error to Cloudflare? … yes

--- a/.changeset/fifty-crabs-flow.md
+++ b/.changeset/fifty-crabs-flow.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+fix: ensure API failures without additional messages logged correctly

--- a/packages/wrangler/src/__tests__/cfetch-utils.test.ts
+++ b/packages/wrangler/src/__tests__/cfetch-utils.test.ts
@@ -78,4 +78,22 @@ describe("throwFetchError", () => {
 			],
 		});
 	});
+
+	it("should include api errors without messages", async () => {
+		msw.use(
+			rest.get("*/user", (req, res, ctx) => {
+				return res(
+					ctx.json({
+						result: null,
+						success: false,
+						errors: [{ code: 10000, message: "error" }],
+					})
+				);
+			})
+		);
+		await expect(runWrangler("whoami")).rejects.toMatchObject({
+			text: "A request to the Cloudflare API (/user) failed.",
+			notes: [{ text: "error [code: 10000]" }],
+		});
+	});
 });

--- a/packages/wrangler/src/cfetch/index.ts
+++ b/packages/wrangler/src/cfetch/index.ts
@@ -13,7 +13,7 @@ export interface FetchResult<ResponseType = unknown> {
 	success: boolean;
 	result: ResponseType;
 	errors: FetchError[];
-	messages: string[];
+	messages?: string[];
 	result_info?: unknown;
 }
 
@@ -168,7 +168,7 @@ function throwFetchError(
 		text: `A request to the Cloudflare API (${resource}) failed.`,
 		notes: [
 			...response.errors.map((err) => ({ text: renderError(err) })),
-			...response.messages.map((text) => ({ text })),
+			...(response.messages?.map((text) => ({ text })) ?? []),
 		],
 	});
 	// add the first error code directly to this error

